lib/extensions/ar_virtual.rb

```diff
@@ -66,17 +66,36 @@ module ClassMethods
     #
 
     def virtual_delegate(*methods)
-      options = methods.pop
-      unless options.kind_of?(Hash) && options[:to]
+      options = methods.extract_options! || {}
+      unless (to = options[:to])
         raise ArgumentError, 'Delegation needs an association. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter).'
       end
-      delegate(*methods, options)
```

Comment:
Seems like we can remove the uses of `:allow_nil => true` since we are no longer deferring to `delegate`:

https://github.com/ManageIQ/manageiq/blob/ba69b49e/app/models/vm_or_template.rb#L183-L187